src/openrct2/world/MoneyEffect.cpp

```diff
@@ -44,8 +59,10 @@ void money_effect_create_at(money32 value, int32_t x, int32_t y, int32_t z, bool
     int16_t offsetX = 0;
     if (!gOpenRCT2NoGraphics)
     {
-        // Construct string to display
-        rct_string_id stringId = money_effect_get_string_id(moneyEffect, &value);
+        auto stringIdValueTuple = moneyEffect->GetStringId();
+        rct_string_id stringId = stringIdValueTuple.first;
+        money32 newValue = stringIdValueTuple.second;
+        *(&value) = newValue;
```

Comment:
Is this awkward code really necessary?

Reply:
Decomposing the tuple isn't necessary, I was just unfamiliar with structured bindings. That's fixed.

If you're referring to the `*(&value) = newValue;`, that's what the original code essentially does, it changes the "value" of the value inside GetStringId(). Passing by reference is also awkward due to
`CreateAt(-value,...)` being called, so I'm not sure of a better solution.

Reply:
I don't think this is necessary either. The get string id function is basically calling absolute on the value. Just pass newValue into format string. No need to overwrite value. Ideally the parameters of this function should be const as they shouldn't be modified.

Reply:
I looked over the old code and I agree. I missed the fact that the change wasn't propagated up the call stack.